src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -261,14 +276,48 @@ public String getStorageAccountName() {
         return storageAccountName;
     }
 
+    public static String getStorageAccountName(final String type, final String newName, final String existName) {
+        //type maybe null in this version, so we can guess according to whether newName is blank or not
+        if (StringUtils.isBlank(type) && StringUtils.isNotBlank(newName)
+                || StringUtils.isNotBlank(type) && type.equalsIgnoreCase("new")) {
+            return newName;
+        }
+        return existName;
+    }
+
+    public String getStorageAccountNameReferenceType() {
+        return storageAccountNameReferenceType;
+    }
+
     public void setStorageAccountName(final String storageAccountName) {
         this.storageAccountName = storageAccountName;
     }
 
+    public String getNewStorageAccountName() {
+        //backward compatibility, the old version plugin only have storageAccountName
+        if (StringUtils.isBlank(newStorageAccountName) && StringUtils.isBlank(existStorageAccountName)
+                && StringUtils.isNotBlank(storageAccountName)) {
+            newStorageAccountName = storageAccountName;
+        }
+
+        return newStorageAccountName;
+    }
+
+    public String getExistStorageAccountName() {
+        return existStorageAccountName;
+    }
+
     public Node.Mode getUseAgentAlwaysIfAvail() {
         return (usageMode == null) ? Node.Mode.NORMAL : usageMode;
     }
 
+    public String isCreateNewStorageAccount(final String type) {
```

Comment:
This method returns whether to create a new account? Then why we need a parameter called type?